pkgs/by-name/dr/draupnir/package.nix

```diff
@@ -0,0 +1,85 @@
+{
+  lib,
+  buildNpmPackage,
+  fetchFromGitHub,
+  pkg-config,
+  sqlite,
+  fetchpatch,
+  matrix-sdk-crypto-nodejs,
+}:
+
+buildNpmPackage rec {
+  pname = "draupnir";
+  version = "2.0.0-beta.6";
+
+  src = fetchFromGitHub {
+    owner = "the-draupnir-project";
+    repo = "Draupnir";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-s1LWXVwY+7LD7cJtZW7mBLsdpB499zS/nDsJ7qaQDfg=";
+  };
+
+  patches = [ ./remove-yarn.patch ];
+
+  postPatch = ''
+    rm yarn.lock
+    cp ${./package-lock.json} ./package-lock.json
+
+    # so that npmInstallHook picks it up
+    substituteInPlace package.json --replace-fail '"main": "' '"bin": "'
+  '';
+
+  npmDepsHash = "sha256-zKfLnxXo383Fjb+xJWoRyFaMrpftYaFhyqPBe6pQ5Ek=";
+
+  npmFlags = [ "--legacy-peer-deps" ];
+  npmRebuildFlags = [ "--build-from-source" ];
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ sqlite ];
+
+  # TODO: move to preNpmRebuild
```

Comment:
Unsure yet if I'd like to do this in this PR (would decrease build time by a bit), but we'll see after it gets tested.